pkg/services/interfaces.go

```diff
@@ -71,6 +71,9 @@ type NetworkProvider interface {
 	// HasLoadBalancer indicates whether this provider has a load balancer for Services.
 	HasLoadBalancer() bool
 
+	// SupportVMReadinessProbe indicates whether this provider support vm readiness probe.
+	SupportVMReadinessProbe() bool
```

Comment:
I think `SupportsVMRedinessProbe` or `VMReadinessProbeSupported` would be more aligned with general naming convention.